<launch>
  <arg name="launch_prefix" default="" />
  <arg name="camera_name" default="/my_robot/camera1" />


  <param name="robot_description" command="cat '$(find apriltag_localization_simulations)/urdf/car_no_kinect.urdf'"/>

  <!-- send fake joint values -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="use_gui" value="False"/>
  </node>

  <!-- Combine joint values -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find apriltag_localization_simulations)/worlds/competition_v2.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>
  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model my_robot -x .2 -y .2 -z 0.0 -param robot_description" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find apriltag_localization_simulations)/rviz/rviz.rviz"/>


  <node pkg="tf" type="static_transform_publisher" name="camera" args="0.0 0.0 0.0 -1.57 0 -1.57 camera_link camera 100" />
  <!-- Launch the tracking node -->
  <!-- <node pkg="visp_auto_tracker" type="visp_auto_tracker" name="visp_auto_tracker" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="model_path" value="$(find visp_auto_tracker)/models" />
    <param name="model_name" value="pattern" />
    <param name="debug_display" value="True" />
	      
    <remap from="/visp_auto_tracker/camera_info" to="$(arg camera_name)/camera_info"/>
    <remap from="/visp_auto_tracker/image_raw" to="$(arg camera_name)/image_raw"/>
  </node> -->

</launch>



